### ISSUE TITLE
racing backends?

### ISSUE THREAD

**Issue by masse-wg2** [99designs/aws-vault#355]
aws-vault v4.5.1 on ubuntu 18.10
So I see this intermittent error, illustrated below. I do `aws-vault exec ... bash` a number of times as fast as I can. Every 5th time or so it fails.
It seems like there is a race between the backends? I have pass and secret-service available.
Also, the --debug flag does not tell me which backend is picked?

```
$ aws-vault exec --debug test bash
2019/04/04 17:36:17 [keyring] Considering backends: [kwallet secret-service pass file]
2019/04/04 17:36:17 [keyring] Failed backend kwallet: Failed to execute program org.kde.kwalletd: No such file or directory
2019/04/04 17:36:17 Loading config file /home/masse/.aws/config
2019/04/04 17:36:17 Parsing config file /home/masse/.aws/config
2019/04/04 17:36:17 Looking for sessions for test
2019/04/04 17:36:17 Looking up all keys in keyring
2019/04/04 17:36:17 Session "prod session (1554405197)" expires in 3h36m59.572337331s
2019/04/04 17:36:17 Session "test session (1554405585)" expires in 3h43m27.572280842s
2019/04/04 17:36:17 Using session ****************3B63, expires in 3h43m26.568475922s
2019/04/04 17:36:17 Assuming role arn:aws:iam::773304303304:role/developer-wg2 from session token
2019/04/04 17:36:18 Using role ****************5LNR (from session token), expires in 14m59.873958727s
2019/04/04 17:36:18 Setting subprocess env: AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY
2019/04/04 17:36:18 Setting subprocess env: AWS_SESSION_TOKEN, AWS_SECURITY_TOKEN

[test](~:master *%)$ exit

$ aws-vault exec --debug test bash
2019/04/04 17:36:19 [keyring] Considering backends: [pass kwallet secret-service file]
2019/04/04 17:36:19 Loading config file /home/masse/.aws/config
2019/04/04 17:36:19 Parsing config file /home/masse/.aws/config
2019/04/04 17:36:19 Looking for sessions for test
2019/04/04 17:36:19 Looking up all keys in keyring
2019/04/04 17:36:19 Session not found in keyring for test
2019/04/04 17:36:19 Looking up keyring for default
Error: default is not in the password store.
2019/04/04 17:36:19 Error from keyring exit status 1
2019/04/04 17:36:19 Failed to find credentials for profile "default" in keyring
aws-vault: error: Failed to get credentials for default (source profile for test): exit status 1

$ aws-vault exec --debug test bash
2019/04/04 17:37:32 [keyring] Considering backends: [secret-service pass kwallet file]
2019/04/04 17:37:32 Loading config file /home/masse/.aws/config
2019/04/04 17:37:32 Parsing config file /home/masse/.aws/config
2019/04/04 17:37:32 Looking for sessions for test
2019/04/04 17:37:32 Looking up all keys in keyring
2019/04/04 17:37:32 Session "prod session (1554405197)" expires in 3h35m44.226692921s
2019/04/04 17:37:32 Session "test session (1554405585)" expires in 3h42m12.226610279s
2019/04/04 17:37:32 Using session ****************3B63, expires in 3h42m11.219969611s
2019/04/04 17:37:32 Assuming role arn:aws:iam::773304303304:role/developer-wg2 from session token
2019/04/04 17:37:33 Using role ****************6IUY (from session token), expires in 14m59.633718795s
2019/04/04 17:37:33 Setting subprocess env: AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY
2019/04/04 17:37:33 Setting subprocess env: AWS_SESSION_TOKEN, AWS_SECURITY_TOKEN
```

**Comment by sciurus** [comment]
I believe I'm seeing the same thing. I keep aws-vault's secrets in os x's keychain. I also have `pass` setup for something else. Occasionally I get the message `Error: foo is not in the password store.` If I run aws-vault again, it just works.

**Comment by lox** [comment]
This was fixed recently in `keyring`: https://github.com/99designs/keyring/pull/36. We need to bump that lib here to fix.

**Comment by nielslaukens** [comment]
Any news on bumping the `keyring` lib?

**Comment by mtibben** [comment]
This is fixed in v4.6.2
